src/util/irep_hash_container.h

```diff
@@ -55,9 +57,10 @@ class irep_hash_container_baset
   {
     inline size_t operator()(const packedt &p) const
     {
+      constexpr size_t bits=sizeof(size_t)*CHAR_BIT;
       size_t result=p.size();
-      for(unsigned i=0; i<p.size(); i++)
-        result^=p[i]<<i;
+      for(size_t i=0; i<std::min(p.size(), bits); i++)
+        result^=((size_t) p[i])<<i;
```

Comment:
We've got irep_hash.h, which has proper hashing functions; just use `hash_combine` I'd say.